tests/AnonymizeTest.php

```diff
@@ -0,0 +1,280 @@
+<?php
+/**
+ * @author Pavel Goncharov <pgoncharov@higherlogic.com>
+ * @copyright 2009-2022 Vanilla Forums Inc.
+ * @license MIT
+ */
+
+namespace Vanilla\KeenAnonymizer\Tests;
+
+use DateTime;
+use DateTimeInterface;
+use Exception;
+use Garden\Hydrate\Exception\ResolverNotFoundException;
+use Vanilla\KeenAnonymizer\Anonymize;
+use Garden\Http\HttpClient;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Tests for the `Anonymize` class.
+ */
+class AnonymizeTest extends TestCase
+{
+    /**
+     * The version of the API we're using.
+     */
+    const API_VERSION = "3.0";
+
+    /** @var array */
+    private $config = [];
+
+    /** @var string */
+    private $collection = "";
+
+    /** @var Anonymize */
+    private $anonymize;
+
+    /** @var HttpClient */
+    private $client;
+
+    /** @var string */
+    private $userEmail;
+
+    /**
+     * Set up a anonymizer fixture for each test.
+     */
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->client = new HttpClient();
+        $this->config = [
+            "version" => self::API_VERSION,
+            "baseUrl" => "https://api.keen.io/" . self::API_VERSION,
+            "masterKey" => "",
+            "orgID" => getenv("TEST_KEEN_ORGID"),
+            "orgKey" => getenv("TEST_KEEN_ORGKEY"),
+            "projectID" => "",
+            "writeKey" => "",
+            "readKey" => "",
+        ];
+        if ($this->config["orgID"] == "" || $this->config["orgKey"] == "") {
+            throw new Exception("Check orgID/orgKey configs.");
+        }
+        $this->collection = "populateEvents" . time();
+        $this->userEmail = "test@higherlogic.com";
+        if (!$this->createProject($this->collection)) {
+            sleep(5);
+            if (!$this->createProject($this->collection)) {
+                throw new Exception(
+                    "Unable to create project, check orgID/orgKey configs.  {$this->config["orgID"]} / {$this->config["orgKey"]}"
+                );
+            }
+        }
+    }
+
+    /**
+     * Tear down project
+     */
+    protected function tearDown(): void
+    {
+        $this->dropProject($this->config["projectID"]);
+    }
+
+    /**
+     * Test populate keen data, and anonymize it.
+     */
+    public function testAnonymizeSuccess()
+    {
+        $this->anonymize = new Anonymize($this->config);
+        $startDate = new DateTime("now");
+        $startDate->modify("-1year");
+        $insertedElements = $this->populateEvents();
+
+        $filter = [
+            "property_name" => "{$this->collection}.recordUser.userID",
+            "property_value" => [10],
+            "startDate" => $startDate->format(DateTimeInterface::ISO8601),
```

Comment:
```suggestion
            "startDate" => $startDate->format(DateTimeInterface::RFC3339),
```

We tend to user RFC3339 everywhere else.